beacon-chain/sync/rpc_blob_sidecars_by_root.go

```diff
@@ -49,15 +49,21 @@ func (s *Service) blobSidecarByRootRPCHandler(ctx context.Context, msg interface
 	blobIdents := *ref
 	for i := range blobIdents {
 		root, idx := bytesutil.ToBytes32(blobIdents[i].BlockRoot), blobIdents[i].Index
-		sc, err := s.blobs.BlobSidecar(root, idx)
+		scs, err := s.cfg.beaconDB.BlobSidecarsByRoot(ctx, root)
```

Comment:
This is moving things in the wrong direction IMO. I still prefer using a narrow blob db interface.

Reply:
No one uses the index, L2s, beacon-API endpoint definition, and consensus spec for data availability and proof checks. As far as I can see, all the usage requires you to get *everything* or *nothing*

narrow db interface has the downside that is more database transaction is required:
- DB tx to get block for the kzg commitment count
- for 0 .. count
 - DB tx to get the blob with the index
 
 Worst case is 5 database transaction vs 1 database transaction

We certainly can optimize and reduce the number of transaction count while using index but I'm not entirely sure that's worth it given what the consumes are going to do